artifacts/camara-cloudevents/event-subscription-template.yaml

```diff
@@ -0,0 +1,756 @@
+      type: object
+      required:
+        - sink
+        - protocol
+        - config
+      properties:
+        protocol:
+          $ref: "#/components/schemas/Protocol"
+        sink:
+          type: string
+          format: url
+          description: The address to which events shall be delivered using the selected protocol.
+          example: "https://endpoint.example.com/webhook"
+        sinkCredential:
+          $ref: '#/components/schemas/SinkCredential'
+        types:
+          description: Camara Event types eligible to be delivered by this subscription - Use of array MUST be decided at API project level.
+          type: array
+          items:
+            type: string
+        filters:
+          type: object
+          description: Placeholder to add filters management for futur UC. **Not to be used as of now.**
+        config:
+          $ref: '#/components/schemas/Config'
+      discriminator:
+        propertyName: protocol
+
+    Protocol:
+      type: string
+      enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
+      description: Identifier of a delivery protocol. Only HTTP is used for now
+      example: "HTTP"
+
+    Config:
+      description: |
+        Implementation-specific configuration parameters needed by the subscription manager for acquiring events.
+        In CAMARA we have predefined attributes like `subscriptionExpireTime`, `subscriptionMaxEvents`, `initialEvent`
+        Specific event type attributes must be defined in `subscriptionDetail`
+        Note: if a request is performed for several event type, all subscribed event will use same `config` parameters.
+      type: object
+      required:
+        - subscriptionDetail
+      properties:
+        subscriptionDetail:
+            $ref: '#/components/schemas/CreateSubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The subscription expiration time (in date-time format) requested by the API consumer. Up to API project decision to keep it.
+        subscriptionMaxEvents:
+          type: integer
+          description: Identifies the maximum number of event reports to be generated (>=1) requested by the API consumer - Once this number is reached, the subscription ends. Up to API project decision to keep it.
+          minimum: 1
+          example: 5
+        initialEvent:
+          type: boolean
+          description: |
+            Set to `true` by APi consumer if consumer wants to get an event as soon as the subscription is created and current situation reflects event request.
+            Example: Consumer request Roaming event. If consumer sets initialEvent to true and device is in roaming situation, an event is triggered 
+            Up to API project decision to keep it.
+
+    SinkCredential:
+      type: object
+      properties:
+         credentialType:
```

Comment:
@PedroDiez : Do you want that we limit the enum to only ACCESSTOKEN for now? It will be easy to update later if other credential type are introduced. I keep the schema for the other for future use.

Reply:
I understand the idea about having the model ready for the future (as done for protocol). Indeed we are aligned in using so far HTTP and in that way an exception is defined. So keeping the polymorphic schema is not the point to me.

I recognize my thoughts were about applying this model with the current scope we have (Bearer Access Token).

So in case we are align that this is the initial understanding, I think we need:
* Trigger an exception to manage other scenarios not covered, just to only allow ACCESSTOKEN
* Makes Token Type to enum "Bearer" or define an exception, in ordeer to not allow other token types.

Just to avoid scenarios that we are not going to manage so far.

In case your initial view is in allow ACCESSTOKEN widely or even allow other credential Types, I prefer to have a meeting with you and Shilpa. To understand how these credential types are seen to work, so i can check internally and try to reach consensus.

My worring is not having a clear guidance about a procedure that we do have not talked in advance and an API Consumer may think it is supported and enabled